From dbf9f461dfe2a078d442a443908dc2f23340c87f Mon Sep 17 00:00:00 2001
Message-Id: <dbf9f461dfe2a078d442a443908dc2f23340c87f.1440106296.git.chang-joon.lee@intel.com>
In-Reply-To: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
References: <1f426693c14c79e65ad8e62ef6bb4b07d6484040.1440106296.git.chang-joon.lee@intel.com>
From: John Harrison <John.C.Harrison@Intel.com>
Date: Wed, 29 Jul 2015 13:58:47 +0100
Subject: [PATCH 03/11] SQUASHME! [VPG]: drm/i915: Fix sched seqno after
 suspend/resume.
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The scheduler seqno is already reset after TDR, but also needs
to be cleared after suspend/resume. The clear of the cache needs to
be done in init_seqno not just in the scheduler’s TDR path.

Note, there is an argument for removing the clear from the scheduler’s
TDR path on the grounds that it can’t be called without also
having init_seqno being called - only init_seqno can actually clear
the seqno so that’s when we should clear the seqno cache.

SQUASHME! - This patch should be squashed into the following patch:
	Author: John Harrison <John.C.Harrison@Intel.com>
	Date:   Wed Jun 3 12:45:53 2015 +0100
	Change-Id: I2addad6e595c68c97897488e80d73e9a7deb6e91
	SQUASHME! [VPG]: drm/i915: Fix for IRQ seqno cache clashing with TDR

Tracked-On: https://jira01.devtools.intel.com/browse/VIZ-4741
Change-Id: Iec3ba7e21d8a90857034e79041bde3db7a0f1a29
Signed-off-by: Peter Lawthers <peter.lawthers@intel.com>

## Don't forget to add a JIRA URL to the Tracked-On metadata!
## https://wiki.ith.intel.com/display/CACTUS/Patch+requirements+check
## url should be in the form:
## Tracked-On: https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
## Several Tracked-On lines are allowed
## you can fully prefill your Tracked-On commits using environment variable:
## export CURRENT_TRACKED_ON=https://jira01.devtools.intel.com/browse/<PROJ>-<NUMB>
##
---
 drivers/gpu/drm/i915/i915_gem.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 9ae8156..a93bffb 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -2580,6 +2580,10 @@ i915_gem_init_seqno(struct drm_device *dev, u32 seqno)
 			ring->semaphore.sync_seqno[j] = 0;
 	}
 
+	if (dev_priv->scheduler)
+		memset(dev_priv->scheduler->last_irq_seqno, 0x00,
+			sizeof(dev_priv->scheduler->last_irq_seqno));
+
 	return 0;
 }
 
-- 
1.9.1

